### PR TITLE
metadata: always initialize metadata, avoid overwriting migration

### PR DESCRIPTION
**Pull request by jbohren**
This fixes an issue where a user can lose workspace configuration when migrating from 0.3.1 to 0.4.0.

Also fixes:
- #319

### CODE REVIEW

File: catkin_tools/metadata.py [hunk 1]
```diff
@@ -152,23 +174,28 @@ def migrate_metadata(workspace_path, old_version):
...
+                isolate_devel = metadata.get('isolate_devel')
+                if isolate_devel is not None:
                     del metadata['isolate_devel']
-                metadata['devel_layout'] = ('isolated' if isolated_devel else 'merged')
+                devel_layout = ('isolated' if isolate_devel else 'merged')
+                metadata['devel_layout'] = devel_layout
 
             # Save the new metadata
-            update_metadata(workspace_path, profile_name, verb, metadata, no_init=True)
+            new_metadata = update_metadata(workspace_path, profile_name, verb, metadata, no_init=True, merge=False)
```
Comment (wjwwood) [inline catkin_tools/metadata.py]: `new_metadata` is unused, is that correct?

Comment (wjwwood) [inline catkin_tools/metadata.py]: Also, I think this line is too long (which is failing the CI).

**Comment by jbohren** [comment]
@NikolausDemmel This should fix the migration problem you saw.

**Comment by jbohren** [comment]
@wjwwood This should be good to merge, I cleaned up the leftover stuff from debugging that was in the earlier commit.

**Pull request merged**